modules/images/dominant-color/class-wp-dominant-color.php

```diff
@@ -0,0 +1,383 @@
+<?php
+/**
+ *  Add Dominant Color preload support to WordPress.
+ *
+ * @package performance-lab
+ * @since 1.0.0
+ */
+
+/**
+ * Add Dominant Color preload support to WordPress.
+ *
+ * @since 1.0.0
+ */
+class WP_Dominant_Color {
+
+	/**
+	 * Class constructor.
+	 */
+	public function __construct() {
+
+		add_filter( 'wp_enqueue_scripts', array( $this, 'add_inline_style' ) );
+
+		add_filter( 'wp_generate_attachment_metadata', array( $this, 'dominant_color_metadata' ), 10, 2 );
+		add_filter( 'wp_generate_attachment_metadata', array( $this, 'has_transparency_metadata' ), 10, 2 );
+		add_filter( 'wp_content_img_tag', array( $this, 'tag_add_adjust' ), 20, 3 );
+		add_filter( 'wp_dominant_color_img_tag_add_adjust', array( $this, 'tag_add_adjust' ), 10, 3 );
+
+		$filters = array( 'the_content', 'the_excerpt', 'widget_text_content', 'widget_block_content' );
+		foreach ( $filters as $filter ) {
+			add_filter( $filter, array( $this, 'filter_content_tags' ), 20 );
+		}
+
+		add_filter( 'wp_get_attachment_image_attributes', array( $this, 'tag_add_adjust_to_image_attributes' ), 10, 2 );
+	}
```

Comment:
Adding hooks in a constructor is not a great pattern, it means unexpected side effects just from instantiating a class. This problem confirms my suggestion above that we should follow core's default pattern and use primarily functions.